pmlc/conversion/gpu_to_spirv/gpu_to_spirv.cc

```diff
@@ -31,13 +32,31 @@ struct StdxSubgroupBroadcastOpConversion
     auto stdxType = op.getResult().getType();
     auto spirvType = typeConverter.convertType(stdxType);
     rewriter.replaceOpWithNewOp<spirv::GroupBroadcastOp>(
-        op, spirvType, spirv::Scope::Subgroup, op.getOperand(0),
-        op.getOperand(1));
+        op, spirvType, spirv::Scope::Subgroup, operands[0], operands[1]);
 
     return success();
   }
 };
 
+struct AllocOpPattern final : public SPIRVOpLowering<AllocOp> {
+public:
+  using SPIRVOpLowering<AllocOp>::SPIRVOpLowering;
+
+  AllocOpPattern(MLIRContext *context, SPIRVTypeConverter &typeConverter)
+      : SPIRVOpLowering<AllocOp>(context, typeConverter, /*benefit=*/1000) {}
+
+  LogicalResult
+  matchAndRewrite(AllocOp operation, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    MemRefType allocType = operation.getType();
+    MemRefType allocType2 = MemRefType::Builder(allocType).setMemorySpace(6);
```

Comment:
Magic number 6?

Reply:
Yeah, this is just lack of cleanup, need to pull in the appropriate enum (6 is the function local spirv memory space).